noether_gui/src/widgets/tpp_widget.cpp

```diff
@@ -192,6 +254,69 @@ vtkSmartPointer<vtkTransform> toVTK(const Eigen::Isometry3d& mat)
   return t;
 }
 
+vtkAssembly* createToolPathActors(const std::vector<ToolPaths>& tool_paths,
+                                  vtkAlgorithmOutput* waypoint_shape_output_port)
+{
+  auto assembly = vtkAssembly::New();
+  for (const ToolPaths& fragment : tool_paths)
+  {
+    for (const ToolPath& tool_path : fragment)
+    {
+      for (const ToolPathSegment& segment : tool_path)
+      {
+        for (const Eigen::Isometry3d& w : segment)
+        {
+          auto transform_filter = vtkSmartPointer<vtkTransformFilter>::New();
+          transform_filter->SetTransform(toVTK(w));
+          transform_filter->SetInputConnection(waypoint_shape_output_port);
+
+          auto map = vtkSmartPointer<vtkPolyDataMapper>::New();
+          map->SetInputConnection(transform_filter->GetOutputPort());
+
+          auto actor = vtkSmartPointer<vtkActor>::New();
+          actor->SetMapper(map);
+
+          assembly->AddPart(actor);
+        }
+      }
+    }
+  }
+
+  return assembly;
+}
+
+// Function to combine two pcl::PolygonMesh instances
+pcl::PolygonMesh combineMeshes(const pcl::PolygonMesh& mesh1, const pcl::PolygonMesh& mesh2) {
+    pcl::PolygonMesh combinedMesh;
+
+    // Combine vertices
+    pcl::PointCloud<pcl::PointXYZ> combinedCloud;
+    pcl::fromPCLPointCloud2(mesh1.cloud, combinedCloud);
+    pcl::PointCloud<pcl::PointXYZ> cloud2;
+    pcl::fromPCLPointCloud2(mesh2.cloud, cloud2);
+
+    combinedCloud += cloud2;
+
+    pcl::toPCLPointCloud2(combinedCloud, combinedMesh.cloud);
+
+    // Combine faces
+    // Add faces for mesh 1
+    combinedMesh.polygons.insert(
+        combinedMesh.polygons.end(), mesh1.polygons.begin(), mesh1.polygons.end());
+
+    // Offset the indices of the second mesh's faces to account for the combined vertices
+    size_t offset = mesh1.cloud.width * mesh1.cloud.height;
+    for (auto& polygon : mesh2.polygons) {
+        pcl::Vertices newPolygon;
+        for (auto index : polygon.vertices) {
+            newPolygon.vertices.push_back(index + offset);
+        }
+        combinedMesh.polygons.push_back(newPolygon);
+    }
+
+    return combinedMesh;
+}
```

Comment:
Let's double check that `pcl::PolygonMesh` concatenate doesn't work here and report what is causing the issue. If this function is necessary, please use `snake_case` for variable names

Reply:
Turns out that `pcl::PolygonMesh::concatenate` had a bug that was fixed [here](https://github.com/PointCloudLibrary/pcl/pull/4745) between 1.11 and 1.12. Looks like we need the custom function after all.